### PR TITLE
feat: Add webhook deliveries index page for client application V1 users

### PR DESCRIPTION
**Pull request by jpotts244**
Add webhook deliveries index page for client application V1 users which will serve as a log of all webhook deliveries attempted by Artsy to this Client Application

See inline screenshots for more details

## TO DOs:
- [ ] Add a show page for each delivery with full event JSON
- [ ] Maybe display the `webhook.endpoint_url` somewhere so clients can see what endpoint we are attempting to send webhooks to
- [ ] Redeliver functionality
- [ ] Fix pagination (right now I have total pages hardcoded, lets pull this from the API but we have some time!

### CODE REVIEW

File: app/assets/stylesheets/webhook_deliveries.css [hunk 1]
```diff
@@ -0,0 +1,12 @@
+.response-status-success {
```
Comment (jpotts244) [inline app/assets/stylesheets/webhook_deliveries.css:1]: Just to help draw the eye, light shading based on `response_status`


Screenshot 2024-09-19 at 5 06 29 PM

File: app/controllers/webhook_deliveries_controller.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    response = ClientApplicationService.fetch_webhook_deliveries(
+      session[:access_token],
+      client_application_id: client_application_id, page: page, size: size
+    )
+
+    @webhook_deliveries = response.map do |webhook_delivery_data|
+      build_webhook_delivery(webhook_delivery_data)
+    end
+
+    @total_pages = 1 # FIX ME: response.headers["X-Total-Count"]
```
Comment (jpotts244) [inline app/controllers/webhook_deliveries_controller.rb:16]: Right now we dont any webhook deliveries, let alone pages of them
Hardcoding for now, I'll do a fast follow in a separate clean up pass

Comment (jpotts244) [reply app/controllers/webhook_deliveries_controller.rb:16]: Pinned to one in this video just to prove that pagination works correctly when communication with Gravity 

https://github.com/user-attachments/assets/8ac913c6-8c26-46bf-a922-715ee6fc50cf

File: app/views/webhook_deliveries/index.html.haml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    %tbody.applications
+      - @webhook_deliveries.each do |webhook_delivery|
+        %tr
+          %td.event_type= webhook_delivery.webhook_event.name
+          %td.id= webhook_delivery.id
+          %td.response_status{ class: status_class(webhook_delivery.response_status) }= webhook_delivery.response_status
+          %td.error_class= webhook_delivery.error_class
+          %td.created_at= webhook_delivery.created_at && DateTime.parse(webhook_delivery.created_at).strftime("%B %d, %Y")
+          %td.delivered_at= webhook_delivery.completed_at && DateTime.parse(webhook_delivery.completed_at).strftime("%B %d, %Y")
+
```
Comment (jpotts244) [inline app/views/webhook_deliveries/index.html.haml:28]: I think displaying the parent webhook's endpoint url would be helpful so clients can see what endpoint we are attempting to hit

(Will do that in a follow up as we need to adjust the response from Grav)

File: app/views/client_applications/show.html.haml [hunk 4]
```diff
@@ -45,6 +45,10 @@
         %th.text-right Configured Application Partners
         %td#client_application_partners
           = link_to "Link", client_application_client_application_partners_path(@client_application.id)
+      %tr
+        %th.text-right Webhook Deliveries
+        %td#client_application_partners
+          = link_to "Link", client_application_webhook_deliveries_path(@client_application.id)
```
Comment (jpotts244) [inline app/views/client_applications/show.html.haml:51]: Clients can access from their application page!


https://github.com/user-attachments/assets/ab447b1e-1cd2-4544-b3f0-29ebb54f56d1

**Review by jonallured — APPROVED** [review]
LGTM!! 🎉

**Pull request merged**